bitnami/zookeeper/3.9/debian-12/rootfs/opt/bitnami/scripts/libzookeeper.sh

```diff
@@ -597,7 +597,7 @@ zookeeper_healthcheck() {
     local args=()
     local port="$ZOO_PORT_NUMBER"
 
-    if [[ "$ZOO_TLS_CLIENT_ENABLE" = true ]]; then
+    if [[ "$ZOO_TLS_CLIENT_ENABLE" = true ]] && is_boolean_yes "$ZOOKEEPER_ENABLE_AUTH"; then
```

Comment:
The environment variable `ZOOKEEPER_ENABLE_AUTH` does not exist.

I think the variable you are looking for is `ZOO_TLS_CLIENT_AUTH`, which can be set to `none`, `want`, `need`.

So your conditional should be like this: 

```suggestion
    if [[ "$ZOO_TLS_CLIENT_ENABLE" = true && "$ZOOKEEPER_ENABLE_AUTH" != "none" ]]; then
```